### PR TITLE
Specify allowed Macmini models

### PR DESCRIPTION
**Pull request by zanderso**
This excludes Macmini7,1.

Fixes https://github.com/flutter/flutter/issues/124877

### CODE REVIEW

**Comment by zanderso** [comment]
It looks like a string is expected, not a map:
```
FormatException: Protobuf JSON decoding failed at: root["platform_properties"]["mac"]["properties"]["dimensions"]. Expected String value
```

**Comment by godofredoc** [comment]
> It looks like a string is expected, not a map:
> 
> ```
> FormatException: Protobuf JSON decoding failed at: root["platform_properties"]["mac"]["properties"]["dimensions"]. Expected String value
> ```

The error is coming from the .ci.yaml proto https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/model/ci_yaml/target.dart;l=40, seems like some changes are going to be needed in the cocoon scheduler.

File: .ci.yaml [hunk 1]
```diff
@@ -55,6 +55,9 @@ platform_properties:
       cpu: x86
       os: Mac-12
       xcode: 14a5294e # xcode 14.0 beta 5
+      dimensions:
+        mac_model: >-
+          ["Macmini8,1", "Macmini9.1"]
```
Comment (godofredoc) [inline .ci.yaml:60]: It seems like `mac_model: "Macmini8,1|Macmini9.1"` would do the trick

File: .ci.yaml [hunk 2]
```diff
@@ -55,6 +55,8 @@ platform_properties:
       cpu: x86
       os: Mac-12
       xcode: 14a5294e # xcode 14.0 beta 5
+    dimensions:
+      mac_model: "Macmini8,1|Macmini9.1"
```
Comment (dnfield) [inline .ci.yaml:59]:
```suggestion
      mac_model: "Macmini8,1|Macmini9,1"
```

**Comment by godofredoc** [comment]
Seems like things are working as expected: 
image

**Review by godofredoc — APPROVED** [review]
Thanks for the change!

**Comment by zanderso** [comment]
👍🏻 No worries. It would be great if this could be self-service, but I think the documentation needs a lot of improvements before that's realistic.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Comment by ricardoamador** [comment]
@zanderso are you referring to the ciyaml readme? If so I can open a follow up task to add more detail.

**Comment by zanderso** [comment]
> @zanderso are you referring to the ciyaml readme? If so I can open a follow up task to add more detail.

Yeah, it would be great to get something like a schema for the file, what fields can go where, what there types and values can be, etc..

**Comment by ricardoamador** [comment]
Okay cool! I will open an infra ticket for that.

**Comment by jmagman** [comment]
You could also remove the `Macmini8,1` dimensions on this change.  Example:
https://github.com/flutter/engine/blob/4ce42fed490418d13da5b5b48b8e36e583dc0ec5/.ci.yaml#L342

**Comment by jmagman** [comment]
> You could also remove the `Macmini8,1` dimensions on this change. Example:
> 
> https://github.com/flutter/engine/blob/4ce42fed490418d13da5b5b48b8e36e583dc0ec5/.ci.yaml#L342

On it: https://github.com/flutter/engine/pull/41223

**Pull request merged**